x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/elasticsearch_monitor_states_adapter.ts

```diff
@@ -29,18 +29,151 @@ interface LegacyMonitorStatesQueryResult {
   afterKey: any | null;
 }
 
+interface LegacyMonitorStatesRecentCheckGroupsQueryResult {
+  checkGroups: string[];
+  afterKey: any | null;
+}
+
+const legacyStatesQuerySize = 100;
```

Comment:
I lowered this to improve page render times. With lots of monitors the page really struggles. I'm debating lowering it again to 50. We should address the react perf in a follow-up PR.